### ISSUE TITLE
Incorrect hostname during install when creating initial email in format of user@box.domain.com

### ISSUE THREAD

**Issue by alento-group** [mail-in-a-box/mailinabox#1515]
When running the install script, it used to be (if my memory serves me correctly) that when the initial email address was chosen that it would end up being user@box.domain.com.  Now I see that when the initial email address is chosen it shows as entered, i.e. user@domain.com.  I assume that this behaviour was modified somewhere along the way .... 

Being a long time user, I create my initial account as miab@box.domain.com as I really want to keep all the MiaB maintenance stuff out of my main email addount.  The unintended consequence of this is that when the box later prompts for the host name is autofills box.box.domain.com.  I can imagine this causing issues for the initiated.  Perhaps a check can be added that if the initial email address has a subdomain, to use the subdomain rather than box.subdomain ...

**Comment by zatricky** [comment]
Sometimes the actual domain is technically a subdomain. It might be a good idea to not make new automatic assumptions.

**Comment by alento-group** [comment]
Exactly!  Now MiaB is making that automatic assumption.... it was better the way it was before.

**Comment by JoshData** [comment]
Tbh I don't think it has changed in many years.

**Comment by alento-group** [comment]
@zatricky 

> Sometimes the actual domain is technically a subdomain. It might be a good idea to not make new automatic assumptions.

Ok, I just ran into a major problem with this when it comes to issuing Let's Encrypt SSL certificates when MiaB is configured only for email on a subdomain.  i.e.  user@box.example.com with no users for user@example.com.  What happens is that the LE client looks for a zone file for example.com for a CAA record.  When it does not find that zone, it causes an error.

https://community.letsencrypt.org/t/another-caa-record-failure-issue/86388
https://github.com/jrsupplee/mailinabox/issues/1
